disk/block.go

```diff
@@ -11,9 +11,40 @@ import (
 var (
 	crc32cTable       = crc32.MakeTable(crc32.Castagnoli)
 	crc32Len          = int64(4)
+	// ErrBadPayloadSize indicates the input payload size is invalid
 	ErrBadPayloadSize = errors.New("disk: bad payload size")
+	// ErrInvalidCRC indicates there is not CRC can be found in the block
+	ErrInvalidCRC = errors.New("disk: not a valid CRC")
 )
 
+// ReadBlock reads a full block into p from rs. len(p) must be (bs - crc32Len).
+// The checksum of the block is calculated and verified.
+// ErrInvalidCRC is returned if the checksum is invalid.
+func ReadBlock(rs io.ReadSeeker, p []byte, index, bs int64 ) error {
+	if int64(len(p)) != bs-crc32Len {
+		return ErrBadPayloadSize
+	}
+	buf := make([]byte, bs)
```

Comment:
do we need this buffer? 

1. read out crc32 to a tmp crc buf
2. readfull from rs.read to p